core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -222,20 +233,26 @@ private void stopContainer(String containerId, String imageName) {
             LOGGER.trace("Was going to stop container but it apparently no longer exists: {}", containerId);
             return;
         } catch (DockerException e) {
-            LOGGER.trace("Error encountered when checking container for shutdown (ID: {}) - it may not have been stopped, or may already be stopped: {}", containerId, e.getMessage());
+            LOGGER.trace(
+                    "Error encountered when checking container for shutdown (ID: {}) - it may not have been stopped, or may already be stopped: {}",
+                    containerId, e.getMessage());
             return;
         }
 
         if (running) {
             try {
                 LOGGER.trace("Stopping container: {}", containerId);
-                dockerClient.killContainerCmd(containerId).exec();
+                dockerClient.stopContainerCmd(containerId).exec();
```

Comment:
it is important to keep the `kill` behaviour here, why do you need to change it to `stop`?

Reply:
the purpose is to stop the container in a grateful way.